src/worker/activities/local_activities.rs

```diff
@@ -135,17 +152,61 @@ impl LocalActivityManager {
         }
     }
 
-    /// Mark a local activity as having completed. Returns the information about the local activity
-    /// so the appropriate workflow instance can be notified of completion.
-    pub(crate) fn complete(&self, task_token: &TaskToken) -> Option<LocalInFlightActInfo> {
-        let info = self
+    /// Mark a local activity as having completed (pass or fail)
+    pub(crate) fn complete(
+        &self,
+        task_token: &TaskToken,
+        status: &LocalActivityExecutionResult,
+    ) -> LACompleteAction {
+        if let Some(info) = self
             .dat
             .lock()
             .outstanding_activity_tasks
-            .remove(task_token)?;
-        self.semaphore.add_permits(1);
-        self.complete_notify.notify_one();
-        Some(info)
+            .remove(task_token)
+        {
+            self.semaphore.add_permits(1);
+
+            match status {
+                LocalActivityExecutionResult::Completed(_) => {
+                    self.complete_notify.notify_one();
+                    LACompleteAction::Report(info)
+                }
+                LocalActivityExecutionResult::Failed(_) => {
+                    match &info.la_info.schedule_cmd.retry_policy {
+                        None => {
+                            // If there's no retry policy... don't retry
+                            LACompleteAction::Report(info)
+                        }
+                        Some(rp) => {
+                            if let Some(backoff_dur) =
+                                rp.should_retry(info.attempt as usize, "TODO")
+                            {
+                                // Send the retry request after waiting the backoff duration
+                                let send_chan = self.act_req_tx.clone();
+                                tokio::spawn(async move {
```

Comment:
There's nothing to cancel this task for example on shutdown, I don't know if I'd leave it dangling like this.

Reply:
I don't think it's relevant at all here. If shutdown is ongoing it cannot proceed until the LAs are complete anyway. If somehow we've decided that's true then this just gets dropped on the floor which seemingly is what you'd want at that point anyway.

Reply:
In the near future we should move shutdown grace period from TS into core and then you'll need to forcefully shut these down.

Reply:
It still won't matter? Nothing will happen. They'll just be dropped. They won't hang shutdown like a thread would.